SharedLibs/Entities/Player/Player.cpp

```diff
@@ -2,28 +2,21 @@
 // Created by hippolyteb on 11/25/16.
 //
 
-#include <iostream>
-#include <thread>
 #include "Player.hpp"
-#include <PartitionSystem/EntityPartitionBuilder.hpp>
-#include <PartitionSystem/Tween/Curve/EaseInOutCurve.hpp>
-#include <PartitionSystem/Tween/Curve/EaseOutCurve.hpp>
+#include <Messages/FireProjectileMessage.hpp>
 
-Player::Player(Timer *timer) : _timer(timer), _partition(timer) {
-    _partition = EntityPartitionBuilder(timer).AddSegment(
-                    PartitionSegmentBuilder()
-                            .Begins(_timer->getCurrent())
-                            .For(std::chrono::seconds(5))
-                            .From(vec2<int>(0, 0))
-                            .To(vec2<int>(500, 500))
-                            .WithCurving(new EaseInOutCurve()))
-            .ContinueWith(PartitionSegmentBuilder()
-                                  .From(vec2<int>(500, 500))
-                                  .For(std::chrono::seconds(10))
-                                  .To(vec2<int>(100, 90))
-                                  .WithCurving(new EaseOutCurve()))
-            .Loop(2)
-            .Build();
+Player::Player(const std::initializer_list<void *> init) : Player(*GetParamFromInitializerList<uint16_t  *>(init, 0), GetParamFromInitializerList<Timer*>(init, 1), GetParamFromInitializerList<RType::EventManager*>(init, 2), *GetParamFromInitializerList<TimeRef*>(init, 3), *GetParamFromInitializerList<vec2<float>*>(init, 4)) { }
+
+Player::Player(uint16_t id, Timer *timer, RType::EventManager *eventManager, TimeRef const &timeRef, vec2<float> const &startPosition) : Entity(id, timer, eventManager) {
+
+}
+
+void Player::Cycle() {
+//    auto now = _timer->getCurrent();
```

Comment:
Because our SharedLibs are being used for test purpose and it would not make sense to create new ones for each tests during dev. They'll be cleaned when we are feature-complete.